_posts/2018-09-13-API-Gateway-Types-Compared.md

```diff
@@ -0,0 +1,153 @@
+---
+layout: post
+title: AWS API Gateway types, use cases and performance
+banner: /assets/posts/2018-09-13-API-Gateway-Types-Compared/banner.png
+author: lvandonkersgoed
+
+---
+API Gateway is a service that allows you to manage access to all sorts of backend systems. Since its release in 2015, many new features and variants have been added. In this post we'll explore the differences, use cases and performance of the Edge Optimized, Regional and Private API Gateway.
+
+# API Gateway use cases
+API Gateway does what its name implies: it's an API that functions as a gateway to upstream services. Its primary use cases are:
+* Allowing HTTP access to services without a native HTTP interface, such as Lambda, Kinesis, Rekognition and many more
+* Versioning different APIs, allowing for backwards compatibility
+* Centrally monitoring and managing traffic coming in to multiple services
+* Presenting a unified API frontend for multiple services
+* Throttling; in general or for specific users
+* Authentication; through Cognito or your own identication provider
+
+# General architecture
+AWS API Gateway resides in an AWS-managed environment. As one of the Serverless services, Amazon manages its hosting, redundancy, scaling, patching, and so on. This means that the underlying technology is completely invisible - all you get is a management interface (Web console, AWS API or AWS CLI). However you choose to configure your API, it will be hosted by AWS. If you want to learn more about Serverless, check out my blog post on [Building a Slack Bot with Serverless Framework](/2018/08/16/Building-a-Slackbot-with-Serverless-Framework.html).
+
+Because a picture speaks a thousand words, here is a visualisation:
+![Architecture](/assets/posts/2018-09-13-API-Gateway-Types-Compared/architecture.png)
+
+# The original API Gateway: Edge Optimized
+At its release in July 2015 (read the [announcement](https://aws.amazon.com/blogs/aws/amazon-api-gateway-build-and-run-scalable-application-backends/)), API Gateway allowed access to Lambda and publicly available HTTP endpoints. At that time, Lambdas could not be placed in VPCs - this feature was [released](https://aws.amazon.com/blogs/aws/new-access-resources-in-a-vpc-from-your-lambda-functions/) in February 2016. In its initial version, API Gateway came paired with a CloudFront distribution. The combined API Gateway and CloudFront would later be called **Edge Optimized** API Gateway, refering to the Edge locations available in CloudFront.
```

Comment:
`Lambdas could not be placed in VPCs` -> `AWS Lambda could not be placed inside a VPC`